browser-switch/src/main/java/com/braintreepayments/api/BrowserSwitchClient.java

```diff
@@ -106,23 +105,30 @@ private boolean isValidRequestCode(int requestCode) {
 
     /**
      * Parses and returns a browser switch result if a match is found for the given {@link BrowserSwitchRequest}
-     * @param pendingRequest the {@link BrowserSwitchPendingRequest.Started} returned from
-     * {@link BrowserSwitchClient#start(ComponentActivity, BrowserSwitchOptions)}
-     * @param intent the intent to return to your application containing a deep link result from the
-     *               browser flow
-     * @return a {@link BrowserSwitchResult.Success} if the browser switch was successfully
-     * completed, or {@link BrowserSwitchResult.NoResult} if no result can be found for the given
-     * {@link BrowserSwitchPendingRequest.Started}. A {@link BrowserSwitchResult.NoResult} will be
+     *
+     * @param intent              the intent to return to your application containing a deep link result from the
+     *                            browser flow
+     * @param pendingRequestState the {@link BrowserSwitchStartResult.Success} token returned from
```

Comment:
I'm not sure about the state/token naming here, those don't align in my mind. Maybe `startResultDetails` or `startResultInfo` or `storedStartResult` and remove `token` from the description?

Reply:
I agree the naming can be improved. Choosing the right names for Browser Switch will maximize the Developer Experience and influence the DX in core.

For me the only thing with "details" and "info" that's conflicting is an implication that the string we return can provide information at runtime. Technically it's an [opaque string](https://stackoverflow.com/a/5838979) that we don't want merchants to inspect.

We should definitely remove the word token, that's a holdover from what I called it originally. But in a way this "state" is similar to a BT nonce–we don't expect merchants to inspect the value for details because the structure of the data is undefined i.e. doesn't follow semantic versioning.

Could we maybe shorten it to `requestState`? Or continue to ideate as a group this might be something to get some 👀's on. I'd be interested in seeing how this bubbles up to the features in the Core SDK too that could help us pick the right names.

Reply:
I think `storedStartResult` makes sense. "result" seems generic enough for a merchant to not want to inspect its value.

Reply:
I like `storedStartResult` for this method. The only thing where the naming doesn't scale as nicely is when the result is [unpacked](https://github.com/braintree/browser-switch-android/blob/3b8bf6256dde8580966094763d1ac8503714ee71/demo/src/main/java/com/braintreepayments/api/demo/ComposeActivity.kt#L74), we'd end up with `result.storedStartResult`.

I'm wondering now if we should just go with something like `originalRequest` to keep it simple? It'll still be opaque because the type is `String`, and it would work nicely in both contexts (`start()` and `parse()`).

Also reiterating the main reason we make this value a `String` is because strings work nicely with all Android persistence mechanisms e.g. SharedPrefs, DataStore, etc. We only need the merchant to hold on to this value for us to remove the responsibility of state restoration out of the SDK. We've gotten a good number of inbounds on that in the past, which makes us think that state restoration logic may be unique to each individual merchant app.

Reply:
The first `result` would be the What if we had separate naming for the `start()` and `parse()` steps? It might give a better description for each for each step. 

Start - `startResultToStore`
``` kotlin
        when (val result = browserSwitchClient.start(this, browserSwitchOptions)) {
            is BrowserSwitchStartResult.Success ->
                PendingRequestStore.put(this, result.startResultToStore)
            ...
        }
```

Parse - `storedStartResult`
```kotlin
       PendingRequestStore.get(this)?.let { storedStartRequest ->
            when (val result = browserSwitchClient.parseResult(intent, storedStartResult)) {
                ...
            }
        }
```

Reply:
Honestly I still kind of think the `pendingRequest` naming is a little more clear for what it is - a request that hasn't been completed yet. Maybe we could change the types to `Started` and `Failure` to better indicate that the request hasn't completed successfully yet? 

```
   when (val pendingRequest = browserSwitchClient.start(this, browserSwitchOptions)) {
            is BrowserSwitchPendingRequest.Started ->
                PendingRequestStore.put(this, pendingRequest.requestToStore)
            ...
        }
```

Also maybe something like `completeRequest` would make more sense than `parse` to indicate what is required by the merchant?

```
       PendingRequestStore.get(this)?.let { storedPendingRequest ->
            when (val result = browserSwitchClient.completeRequest(intent, storedPendingRequest)) {
                ...
            }
        }
```

Can't decide what would be most clear for merchants. Tagging in @scannillo @jaxdesmarais @saperi22 for more opinions

Reply:
PR for renaming `parseResult()` to `completeRequest()`: https://github.com/braintree/browser-switch-android/pull/101